pkg/agent/vhd/cache/cache.go

```diff
@@ -0,0 +1,134 @@
+// Package cache provides types and functionality for reasoning about the content cached on a particular VHD version
+// primary through the usage of both the components.json and manifest.json file artifacts.
+package cache
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/Azure/agentbaker/parts"
+)
+
+const (
+	manifestFilePartPath   = "linux/cloud-init/artifacts/manifest.json"
+	componentsFilePartPath = "linux/cloud-init/artifacts/components.json"
+)
+
+//nolint:gochecknoinits
+func init() {
+	if err := initVHDCacheContent(); err != nil {
+		panic(err)
```

Comment:
Should we use panic or just returning error here? Does the panic occur during VHD baking? It won't crash during node provisioning, right?

Reply:
we panic since this is in an `init()` - https://tutorialedge.net/golang/the-go-init-function/

Reply:
Just to clarify a bit, who is importing/using this cache package?

Reply:
the `agent` pkg is the primary consumer, which contains the bakerapi code